lib/model.js

```diff
@@ -1182,6 +1182,22 @@ class Model {
         // Assign an auto-generated name to indexes which are not named by the user
         this.options.indexes = this.QueryInterface.nameIndexes(this.options.indexes, this.tableName);
 
+        const existedIndexes = {};
+        const duplicatedIndexes = [];
+
+        indexes.forEach(index => {
+          const fieldNameStr = index.fields.map(field => field.attribute).join('_');
+          const key = `${index.type}_${index.primary}_${index.unique}_${fieldNameStr}`;
```

Comment:
`nameIndexes` method also doing something similar, to keep names consistent we should share this code